### ISSUE TITLE
excluding facet counts if the values is 0

### ISSUE THREAD

**Issue by hah** [meilisearch/MeiliSearch#1064]
creating a feature request for: https://github.com/meilisearch/MeiliSearch/pull/1053 it is not a super important feature, more like a nice to have 

**Describe the solution you'd like**
if you have a lot of tags, the results is messy so would be nice to have a feature to only include the "relevant" results (like count > 0)

**Describe alternatives you've considered**
well technically i can just process the results on my end, however:
1) handling it on the server (search engine) feels cleaner
2) in slower languages it would take more time
3) the smaller json saves data

**Comment by MarinPostma** [comment]
with @meilisearch/core-team we don't think that this is something we want. This makes to API more complicated, and since the payloads are compressed, and should not contain too much values, application side sorting should not be too expensive.

that being said, @meilisearch/integration-team may have another opinion on this, and if they do we will reconsider.

**Comment by tpayet** [comment]
Thank you @hah for writing this issue :)

As @MarinPostma exposed it, we are very picky when it comes to adding new settings to the engine.
We have to define whether or not we are ready to add every possible options to the search engine while knowing it may confuse users. 

As you may know we are trying to differenciate from Elastic by providing the simplest developer experience so adding this kind of option that could be handled on the front-end is very complicated for us today. 

We should have a discussion regading settings and how to design them to be the easiest possible. That said, if we decide someday to rework on settings your work will be taken into acount and brought to the discussion. Thanks again!

**Comment by curquiza** [comment]
I agree with @MarinPostma: this would add complexity to the API for a thing that could be easily done in front-end.

On the other hand, the integration tools we provide (meilisearch-js, instant-meilisearch, meilisearch-swift...) could handle this kind of behavior, for example, an option `displayZeroFacets` sets to `true` by default that the users could change to `false`.

For people using instant-meilisearch in front-end, we currently provide a solution (see: https://github.com/meilisearch/instant-meilisearch/issues/126). But a parameter (like `displayZeroFacets`) could also be implemented for more convenience!

TLDR; from my POV the integration tools should handle it, not the core.

**Comment by hah** [comment]
alright, thanks a lot!
